### PR TITLE
docs: Add comprehensive README

### LINKED ISSUE #40: Add a basic README
Describe the project, needed help, etc

### PR DESCRIPTION
**Pull request by vincent067**
## Summary 📝

This PR adds a comprehensive README.md to help new users and contributors understand the project. The repository currently lacks documentation, which makes it difficult for people to get started with the emulator testing framework.

## What's included 👍

- **Project overview** - Clear explanation of what GBEmulatorShootout does
- **Supported emulators** - Complete list of all 17 emulators with links
- **Test ROM documentation** - Lists all test suites used for accuracy testing
- **Installation guide** - Step-by-step setup instructions
- **Usage examples** - Common CLI commands with explanations
- **Output format** - What to expect from test runs
- **Project structure** - Overview of the codebase organization

## Closes

Fixes #40

## Testing 🧪

- Verified all emulator URLs are valid
- Checked that markdown renders correctly
- Confirmed installation steps work on clean environment

Let me know if you'd like any changes! Happy to iterate on this. 😊

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+| [GambatteSpeedrun](https://github.com/pokemon-speedrunning/gambatte-speedrun) | github.com/pokemon-speedrunning/gambatte-speedrun |
+| [Emulicious](https://emulicious.net/) | emulicious.net |
+| [Goomba](https://www.dwedit.org/gba/goombacolor.php) | dwedit.org/gba/goombacolor |
+| [binjgb](https://github.com/binji/binjgb) | github.com/binji/binjgb |
+| [PyBoy](https://github.com/Baekalfen/PyBoy) | github.com/Baekalfen/PyBoy |
+| [ares](https://ares-emu.net/) | ares-emu.net |
+| [Emmy](https://emmy.n1ark.com/) | emmy.n1ark.com |
+| [gameroy](https://github.com/Rodrigodd/gameroy) | github.com/Rodrigodd/gameroy |
+| [DocBoy](https://github.com/Docheinstein/docboy) | github.com/Docheinstein/docboy |
+
```
Comment (ISSOtm) [inline README.md:34]: The list of emulators is already present in the website itself, so this list would easily fall out of sync and thus become problematic; I'd rather it be absent from the README. Perhaps the README's introduction line can evoke the variety of listed emulators?

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
+# GBEmulatorShootout 🎮
+
+A comprehensive comparison framework for testing Game Boy emulator accuracy across hundreds of test ROMs.
+
+## Overview
+
+GBEmulatorShootout automatically tests multiple Game Boy emulators against a suite of accuracy test ROMs, generating detailed HTML reports with pass/fail status and screenshots for each test case.
+
+![License](https://img.shields.io/badge/license-MIT-blue.svg)
+
```
Comment (ISSOtm) [inline README.md:10]: Hmm, I don't think the license is particularly relevant to this project, since it's not a distributable; I'd rather have it only mentioned at the bottom.


```suggestion
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# GBEmulatorShootout 🎮
+
+A comprehensive comparison framework for testing Game Boy emulator accuracy across hundreds of test ROMs.
+
+## Overview
+
```
Comment (ISSOtm) [inline README.md:6]: This is redundant with the sentence below, so I'd like to keep only one. And, personally, I find the latter more descriptive of the project :)

```suggestion
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+GBEmulatorShootout automatically tests multiple Game Boy emulators against a suite of accuracy test ROMs, generating detailed HTML reports with pass/fail status and screenshots for each test case.
```
Comment (ISSOtm) [inline README.md:7]:
```suggestion
GBEmulatorShootout tests multiple Game Boy emulators against a suite of accuracy test ROMs, compiling all the pass/fail results into a comprehensive table, with screenshots for each test case.

The table can be generated locally, but we also regularly update it at https://gbdev.io/GBEmulatorShootout/
```

Stylistic tweaks, and placing a link to the site prominently in the README itself.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+- **Blargg's tests** - Classic Game Boy test ROMs
+- **Mooneye tests** - GB/GBC accuracy tests
+- **Acid tests** - GPU/graphics tests
+- **Samesuite tests** - Various test scenarios
+- **ax6 tests** - Additional accuracy tests
+- **daid tests** - Custom test collection
+- **hacktix tests** - Edge case tests
+- **cpp tests** - C++ emulator tests
+- **mealybug tests** - Timing and accuracy tests
```
Comment (ISSOtm) [inline README.md:47]: Can links be added to the repos those are pulled from? Also, the descriptions are fairly vague; IMO, either make this an inline list (“The framework includes test ROMs from sources such as Blagg's tests, Mooneye's suite, the Acid Tests, SameSuite...”), or write descriptions that point out highlights of each test ROM set.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+### Core Requirements
+```
+pyautogui
+pillow
+requests
+pywin32
+selenium
+tqdm
+```
```
Comment (ISSOtm) [inline README.md:59]: This table is redundant with the `requirements.txt` file itself, so please remove it.


```suggestion
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+Install with:
+```bash
+pip install -r requirements.txt
+```
+
+### System Requirements
+
+- Windows (for most emulator automation)
+- Python 3.7+
+- Game Boy emulators installed and configured
```
Comment (ISSOtm) [inline README.md:70]: This should be self-evident, so please remove this bullet point.


```suggestion
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+```
+
+### Export Test Data
+
+Export emulator and test definitions to JSON:
+
+```bash
+python main.py --dump-emulators-json
+python main.py --dump-tests-json
+```
```
Comment (ISSOtm) [inline README.md:127]: This is a good idea in theory, but it's a really verbose way to specify all that. I'd prefer pointing out, in text, that `main.py` is the test runner, and that one can run `python main.py --help` to learn about all its knobs.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+Example emulator specification:
+```python
+{
+    'factory': lambda: _new_instance("emulators.myemu", "MyEmulator"),
+    'keywords': ["MyEmulator", "myemu"],
+    'name': "MyEmulator",
+    'url': "https://myemulator.example.com/",
+}
+```
```
Comment (ISSOtm) [inline README.md:188]: This itself is good information, but I think it would be better moved in a separate `CONTRIBUTING.md` document for better organisation.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+## Contributing
+
+Contributions are welcome! Here are some ways you can help:
+
+- 🐛 Report bugs or inaccuracies
+- ✨ Add support for new emulators
+- 📚 Improve documentation
+- 🧪 Add new test ROMs
+- 🎨 Enhance the HTML report
```
Comment (ISSOtm) [inline README.md:198]: This list is in itself very generic, I'd prefer if it linked to appropriate sections of the `CONTRIBUTING.md` document (mentioned in the previous comment). If the appropriate section doesn't exist, write a new section with a quick summary of how to perform the appropriate contribution kind, and link to the new section from this list.

File: README.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+
+## Acknowledgments
+
+- All the emulator developers for their amazing work
+- The test ROM authors (Blargg, mooneye, and others)
+- The Game Boy development community
+
+---
+
+**Happy testing!** 🕹️ If you find this project useful, please consider giving it a ⭐
```
Comment (ISSOtm) [inline README.md:212]: Begging for GitHub stars is not in the spirit of our project, but thank you for the effort :grin:


```suggestion
```

File: README.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
+# GBEmulatorShootout 🎮
```
Comment (ISSOtm) [inline README.md:1]:
```suggestion
# GBEmulatorShootout 🎮🔫
```

Let's have some fun, shall we? :3

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Hey! Thanks for submitting this.

The style of what you submitted doesn't match our project's spirit, but that's no big deal, the substance is a good thing to start discussing off of. I have a bunch of requests below, but if you have pushback, feel free to state your reasons aloud!

File: README.md (hunk 1 above)
Comment (Rangi42) [reply README.md:34]: The website doesn't really present a convenient "list" of emulators; you have to scroll sideways over a large table and read its column headers. Short lists of the tested emulators and of the test suites are two things I'd want to see in a README.

File: README.md (hunk 2 above)
Comment (Rangi42) [reply README.md:10]: We can also just rely on the fact that a LICENSE file exists and is detected by GitHub to present in its repo sidebar. If we really want to mention the license in the README, I think it should be done as plain text.

File: README.md (hunk 12 above)
Comment (Rangi42) [reply README.md:1]: Let's not encourage even more emojis. :3
```suggestion
# GBEmulatorShootout
```

File: README.md (hunk 4 above)
Comment (Rangi42) [reply README.md:7]: `s/statuses/results/` and `s/obtained/generated/`, but yes.

File: README.md (hunk 1 above)
Comment (Rangi42) [reply README.md:34]: That said, it's weird for the table-style "list" to have visible URLs that *aren't* linked. Linking them would be redundant with the titles, so just remove the visible URLs; anyone can easily hover/click on the titles to see the pages.

File: README.md (hunk 5 above)
Comment (Rangi42) [reply README.md:47]: I'd just remove the descriptions, unless those test suites provide their own which we can base on. And remove the repetitive "tests" suffixes. This can be a list of titles which are linked to their pages, just like the emulator list.

File: README.md [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+- Windows (for most emulator automation)
+- Python 3.7+
+- Game Boy emulators installed and configured
+
+## Usage
+
+### Basic Usage
+
+Run all tests on all configured emulators:
+
```
Comment (Rangi42) [inline README.md:77]: Sub-sub-headings are excessive for such a short document, and redundant with the brief sentences anyway.
```suggestion
Run all tests on all configured emulators:

```

File: README.md [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+│   ├── bdm.py
+│   ├── mgba.py
+│   ├── sameboy.py
+│   └── ...
+└── testroms/            # Test ROM collections
+    ├── blargg/
+    ├── mooneye/
+    ├── acid/
+    └── ...
+```
```
Comment (Rangi42) [inline README.md:162]: This can be moved to an ARCHITECTURE.md file like rgbds does.

File: README.md (hunk 9 above)
Comment (Rangi42) [reply README.md:188]: Agreed. Also there should be a final list item for updating the documentation.

File: README.md (hunk 10 above)
Comment (Rangi42) [reply README.md:198]: That, and the emojis are just noise.

File: README.md (hunk 11 above)
Comment (Rangi42) [inline README.md:212]: The license is clear from the existence of a LICENSE file. The emulators and tests are clearly acknowledged by listing them and linking to them earlier in this document. Overall the tone here is "verbosely friendly customer service voice", not "simple positive but to-the-point README".
```suggestion
```

**Review by Rangi42 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ISSOtm) [reply README.md:34]: This list is not *short*, however, it's comprehensive, which makes it redundant in my eyes. I don't want to try making a shortlist unless we have some criteria to base them off of; and even those may be subject to constant bikeshedding.

As for the website presenting a list of emulators, I have in mind that this should be solved by implementing a filter for emulators, which would incidentally serve as such a list that doesn't require horizontal scrolling. (And such filtering *would* remove the horizontal scrolling, too!)

File: README.md (hunk 5 above)
Comment (ISSOtm) [reply README.md:47]: Well, Blargg's suite *is* generally considered “GB emulator basics”, for example. (Though we may want to present that info on the website as well, more generally.)

File: README.md (hunk 11 above)
Comment (ISSOtm) [reply README.md:212]: It remains common to mention the license in the README, if only because sometimes the LICENSE file doesn't apply to the whole repo (there may be exceptions). I think it's fine to keep one line about it.

File: README.md (hunk 1 above)
Comment (Rangi42) [reply README.md:34]: I'm not sure what you mean by a "shortlist" with "some criteria to base them off of".

My use of "large" meant "you have to scroll horizontally a lot to read all the emulators' table headings, [which is bad UX](https://www.nngroup.com/articles/horizontal-scrolling/)". (It's inevitable for a table, of course, but means that users should not be expected to treat the table headings as a readable list on their own.)

And my use of "short list" was just an adjective, like "short enough to fit on one screen without scrolling"; no implication that it would be short*ened* by some criteria to not include every single emulator we test.

File: README.md (hunk 5 above)
Comment (Rangi42) [reply README.md:47]: I just meant that if certain test suites are self-described as being about audio, SGB, hblank tricks, etc, that's something we'd want to point out.

But if most of the suites are just "basics"/"various"/"accuracy"/"more accuracy"/"additional various", then descriptions aren't doing much for the reader.

E.g. [Blargg's repo](https://github.com/retrio/gb-test-roms) just says "Collection of Game Boy test roms"; [SameSuite](https://github.com/LIJI32/SameSuite) just says "a set of Game Boy and Game Boy Color test ROMs"; etc. Maybe we could distinguish them by pointing out things like "blargg's are from way back in the 2000s" or "SameSuite was written specifically to help develop SameBoy".

File: README.md (hunk 1 above)
Comment (ISSOtm) [reply README.md:34]: It is short enough to fit one a single screen, even a mobile one; however, it is long enough and lacking obvious organisation that I don't see what a user is supposed to glean from it.

Further, as more emulators are going to be added, this list is going to grow beyond that “short” qualifier; IMO it's already *questionably* short.

File: README.md (hunk 12 above)
Comment (vulcandth) [reply README.md:1]: Normally I would agree about the emoji's but I vote we keep ISSOtm's emoji suggestion. Its actually cute.
